app.js

```diff
@@ -2,13 +2,38 @@ const express = require("express");
 const app = express();
 const path = require("path");
 const handlers = require(path.join(__dirname, "events", "handlers.js"));
+const uuid = require("uuid");
+const { AsyncLocalStorage } = require("async_hooks");
+const storage = new AsyncLocalStorage();
+const Logger = require("./logger").createLogger();
+
+const requestIdMiddleware = (req, res, next) => {
+  storage.run(new Map([["requestId", uuid.v4()]]), () => {
+    next();
+  });
+};
+
+app.use(requestIdMiddleware);
+app.use(function (req, res, next) {
+  ctx = {
+    traceID: storage.getStore().get("requestId"),
+  };
+  req.context = ctx;
```

Comment:
why do we need to put context in request if we can grab it from storage almost anywhere

Reply:
It's a question of habit. I'd prefer to keep the context near (inside) the request